src/cmds/pbs_rsub.c

```diff
@@ -696,8 +765,158 @@ main(int argc, char *argv[], char *envp[])
+						if (select_str == NULL) {
+							fprintf(stderr, "pbs_rsub: Out of memory\n");
+							exit(2);
+						}
+						sprintf(select_str, "%s+host=%s:ncpus=%s", tmp_str, bstat->name, ncpus);
+						free(tmp_str);
+					}
+				}
+			}
+		}
+
+		pbs_statfree(bstat_head);	/* free info returned by pbs_stathost() */
+
+		pbs_errno = 0;
+		bstat_head = pbs_statvnode(connect, "", NULL, NULL);
+		if (bstat_head == NULL) {
+			if (pbs_errno) {
+				errmsg = pbs_geterrmsg(connect);
+				if (errmsg != NULL) {
+					fprintf(stderr, "pbs_rsub: %s\n", errmsg);
+				} else {
+					fprintf(stderr, "pbs_rsub: Error (%d) submitting reservation\n", pbs_errno);
+				}
+			} else {
+				fprintf(stderr, "pbs_rsub: No nodes found\n");
+			}
+
+			CS_close_app();
+			exit(pbs_errno);
+		}
+
+		/* here, the execvnodes is crafted from vnodes */
+		hostp = maintenance_hosts;
+		for (;*hostp ;hostp++) {
+			for (bstat = bstat_head; bstat; bstat = bstat->next) {
+
+				for (pattr = bstat->attribs; pattr; pattr = pattr->next) {
+				    if (pattr->resource && strcmp(pattr->resource, "host") == 0)
+					host = pattr->value;
+
+				    if (pattr->resource && strcmp(pattr->name, "resources_available") == 0 && strcmp(pattr->resource, "ncpus") == 0)
+					ncpus = pattr->value;
+				}
+
+				if (strcmp(host, *hostp) == 0 && atoi(ncpus) > 0){
+					if (!execvnodes_str) {
+						/* +10 stands for '(' + ":ncpus=" + ')' + '\0' */
+						execvnodes_str = (char*) malloc(strlen(bstat->name) + strlen(ncpus) + 10);
+						if (execvnodes_str == NULL) {
+							fprintf(stderr, "pbs_rsub: Out of memory\n");
+							exit(2);
+						}
+						sprintf(execvnodes_str, "(%s:ncpus=%s)", bstat->name, ncpus);
+					} else {
+						tmp_str = strdup(execvnodes_str);
+						if (tmp_str == NULL) {
+							fprintf(stderr, "pbs_rsub: Out of memory\n");
+							exit(2);
+						}
+						/* +11 stands for "+(" + ":ncpus=" + ')' + '\0' */
+						execvnodes_str = (char*) realloc(execvnodes_str, strlen(tmp_str) + strlen(bstat->name) + strlen(ncpus) + 11);
+						if (execvnodes_str == NULL) {
+							fprintf(stderr, "pbs_rsub: Out of memory\n");
+							exit(2);
+						}
+						sprintf(execvnodes_str, "%s+(%s:ncpus=%s)", tmp_str, bstat->name, ncpus);
+						free(tmp_str);
+					}
+				}
+			}
+		}
+
+		pbs_statfree(bstat_head);	/* free info returned by pbs_statvnodes() */
+
+		/* add crafted select */
+		if ((i = set_resources(&attrib, select_str, 0, &erp)) != 0) {
+			if (i > 1) {
+				pbs_prt_parse_err("pbs_rsub: illegal -l value\n", select_str,
+					    (int)(erp-select_str), i);
+			} else {
+				fprintf(stderr, "pbs_rsub: illegal -l value\n");
+			}
+
+			CS_close_app();
+			exit(pbs_errno);
+		}
+
+		/* add place=exclhost */
+		if (set_resources(&attrib, "place=exclhost", 0, &erp) != 0) {
+			fprintf(stderr, "pbs_rsub: illegal -l value\n");
+
+			CS_close_app();
+			exit(pbs_errno);
+		}
+
+		extend = strdup("m");
```

Comment:
instead of calling strdup() and checking for it's success - let us use char *extend = "s".

Reply:
OK, strcat() is used now.